samples/sample/src/main/java/com/microsoft/jfr/App.java

```diff
@@ -0,0 +1,67 @@
+package com.microsoft.jfr;
+
+import javax.management.InstanceNotFoundException;
+import javax.management.MBeanServerConnection;
+import java.io.IOException;
+import java.lang.management.ManagementFactory;
+import java.math.BigInteger;
+import java.nio.file.Paths;
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.RecursiveAction;
+
+public class App 
+{
+    private static final ForkJoinPool threadPool = new ForkJoinPool();
+    private static final int range = 100000;
+
+    static class Fibonacci extends RecursiveAction {
+
+        public BigInteger NthFibonacci(int n) {
+            BigInteger[] fibonacciNumbers = new BigInteger[n+2];
+            fibonacciNumbers[0] = BigInteger.ZERO;
+            fibonacciNumbers[1] = BigInteger.ONE;
+            for(int i = 2; i <= n; i++){
+                fibonacciNumbers[i] = fibonacciNumbers[i-1].add(fibonacciNumbers[i-2]);
+            }
+            return fibonacciNumbers[n];
+        }
+
+        @Override
+        protected void compute() {
+            // Do some computation that heats up the CPU
+            for (int i = 1; i<=range; i++){
+                System.out.println( i + "th Fibonacci number: " + NthFibonacci(i).toString());
+            }
+        }
+    }
+
+    public static void main( String[] args )
+    {
+        // Create a connection to the mBean server
+        MBeanServerConnection mBeanServer = ManagementFactory.getPlatformMBeanServer();
+        try {
+            // In order to upload a JFR file from a remote JVM, we need to create a FlightRecorderConnection
+            FlightRecorderConnection flightRecorderConnection = FlightRecorderConnection.connect(mBeanServer);
+            RecordingOptions recordingOptions = new RecordingOptions.Builder().disk("true").build();
+            RecordingConfiguration recordingConfiguration = RecordingConfiguration.PROFILE_CONFIGURATION;
+
+            // Try creating a new recording instance. This doesn't start recording yet.
+            try (Recording recording = flightRecorderConnection.newRecording(recordingOptions, recordingConfiguration)) {
+                // Start recording
+                recording.start();
+
+                threadPool.invoke(new Fibonacci());
```

Comment:
I think something is missing, or I am missing something. How does this use all cores? 

Can you make this sample duration based? By that I mean, make it so the sample runs for _n_ seconds.

Reply:
That's an interesting approach! Surely can implement that and make the number of seconds configurable. But here the `threadPool` figures out how many threads it should use and make use of all the cores.

Reply:
The point of the sample should be more about showing the API than the quality of the JFR file. You can always say in the documentation to set this property to some large value to get a more interesting JFR file.